torch/profiler/profiler.py

```diff
@@ -44,6 +50,40 @@ def supported_activities():
     return torch.autograd._supported_activities()
 
 
+def _optimizer_post_hook(optimizer, args, kwargs):
+    KinetoStepTracker.increment_step("Optimizer")
+
+
+@contextlib.contextmanager
+def _dynolog_handler_context():
+    """
+    The code before the yield statement runs when the context starts before any code from
+    the context is executed. The code after the yield statement runs after the context ends.
+    """
+    global _dynolog_handle
+    if _dynolog_handle is None:
+        _dynolog_handle = True
+    yield
+    _dynolog_handle = None
+
+
+def _profile_using_dynolog(override: bool = False) -> Optional[RemovableHandle]:
+    """
+    To enable tracing via dynolog we register a global optimizer step post
+    hook. Requires the 'KINETO_USE_DAEMON' environment variable to be set.
+    """
+    global _dynolog_handle
+    hook_handle: Optional[RemovableHandle] = None
+    with _dynolog_handler_context():
+        # check if hook has been registered and environment variable is set properly.
+        # the override option is to handle the test cases only.
+        if _dynolog_handle is True and (os.environ.get("KINETO_USE_DAEMON", None) is not None or override is True):
+            hook_handle = register_optimizer_step_post_hook(_optimizer_post_hook)
+            # set _dynolog_handle to False to prevent re-registeration of the hook.
+            _dynolog_handle = False
```

Comment:
This will get overwritten on exit of the context manager above that sets it back to None right?